src/import-data/protected-data-utils.ts

```diff
@@ -1,28 +1,55 @@
-import { cond } from '@terra-ui-packages/core-utils';
 import { isGoogleWorkspace, WorkspaceWrapper } from 'src/libs/workspace-utils';
 
-// This method identifies whether an import source is considered protected data;
-// For now this means pfb imports from AnVIL or Biodata Catalyst.
-export const isProtectedSource = (url: string, filetype?: string): boolean => {
-  if (!url) {
-    return false;
-  }
-  try {
-    const hostname = new URL(url).hostname;
-    const protectedHosts = [
-      'anvil.gi.ucsc.edu',
-      'anvilproject.org',
-      'gen3.biodatacatalyst.nhlbi.nih.gov',
-      'gen3-biodatacatalyst-nhlbi-nih-gov-pfb-export.s3.amazonaws.com',
-      'gen3-theanvil-io-pfb-export.s3.amazonaws.com',
-    ];
-    return cond([!filetype || !url, () => false], [!!filetype && filetype.toLowerCase() !== 'pfb', () => false], () =>
-      protectedHosts.some((host) => hostname.endsWith(host))
-    );
-  } catch (e) {
-    console.error(`Unable to parse url: ${url}`);
+import { ImportRequest } from './import-types';
+
+type ProtectedSource = { type: 'http'; host: string } | { type: 's3'; bucket: string };
+
+// These must be kept in sync with PROTECTED_URL_PATTERNS in import-service.
+// https://github.com/broadinstitute/import-service/blob/develop/app/protected_data.py
+const protectedSources: ProtectedSource[] = [
+  // AnVIL production
+  { type: 'http', host: 'service.prod.anvil.gi.ucsc.edu' },
+  { type: 's3', bucket: 'edu-ucsc-gi-platform-anvil-prod-storage-anvilprod.us-east-1' },
+  // AnVIL development
+  { type: 'http', host: 'service.anvil.gi.ucsc.edu' },
+  //  BioData Catalyst
+  { type: 'http', host: 'gen3.biodatacatalyst.nhlbi.nih.gov' },
+  { type: 's3', bucket: 'gen3-biodatacatalyst-nhlbi-nih-gov-pfb-export' },
+  { type: 's3', bucket: 'gen3-theanvil-io-pfb-export' },
+];
+
+/**
+ * Determine if a PFB file is considered protected data.
+ * */
+export const isProtectedPfbSource = (pfbUrl: string): boolean => {
+  const parsedUrl = new URL(pfbUrl);
```

Comment:
Are we worried about this ever failing to parse?

Reply:
https://github.com/DataBiosphere/terra-ui/pull/4305/commits/99b9f6db2118050790fc21d5f306de02d1164609 adds validation that the URL can be parsed before this step is reached.